### DISCUSSION TITLE
Setting targetHandle from custom edge

### DISCUSSION THREAD

**Discussion by ayoubphy** [xyflow/xyflow#3128] [Q&A]
Hi Guys,

I have this use case where I need to set targetHandle id (using multiple handles) inside a custom edge, but BaseEdge doesn't have targetHandle prop.

any ideas on how to achieve this ?

**Comment by bcakmakoglu** [comment]
What exactly is this use-case supposed to look like?
What's the goal you're trying to achieve here?
Can you elaborate a bit further on that, so we might be able to help you out with whatever you're trying to build 😄

**Comment by ayoubphy** [comment]
I have a custom edge that decides if the source is from the right or left of the target based on sourceX & targetX, and I want to be able to change the target handle dynamically to either left or right.

**Reply by bcakmakoglu** [reply]
If I understand you correctly you want to force the direction the edge takes, not really change the position of the handle, is that right?

You can use `getBezierEdge` and pass `targetPosition` to force the edge into a specific direction.

```ts
  const [edgePath, labelX, labelY] = getBezierPath({
    sourceX,
    sourceY,
    sourcePosition,
    targetX,
    targetY,
    targetPosition,
  });
```

Or are you asking to actually switch the handles position?
If so, you probably need some sharable state that stores which position (top, left, right, bottom) should be at and read that from your custom node to reposition your handle accordingly.
(Sorry I got no example for that on hand 😅)
